File: apps/api/src/routes/events.ts
import { Router, type Request, type Response } from "express";
import { queryEvents } from "../lib/sqlite.js";

export function createEventsRouter(): Router {
  const router = Router();

  router.get("/", (req: Request, res: Response) => {
    try {
      const sinceMs = Number(req.query.sinceMs) || Date.now() - 24 * 60 * 60 * 1000;
      const untilMs = req.query.untilMs ? Number(req.query.untilMs) : undefined;
      const deviceId = req.query.deviceId as string | undefined;
      const eventType = req.query.eventType as string | undefined;
      const limit = req.query.limit ? Number(req.query.limit) : 100;

      const events = queryEvents({ sinceMs, untilMs, deviceId, eventType, limit });
      res.json({ ok: true, events });
    } catch (err) {
      console.error("Error querying events", err);
      res.status(500).json({ ok: false, error: "Failed to query events" });
    }
  });

  return router;
}

File: apps/cortex/src/api/events.py
"""
Event routes: GET /api/events

Ported from apps/api/src/routes/events.ts
"""

import time

from fastapi import APIRouter, Query


def create_events_router(sqlite) -> APIRouter:
    r = APIRouter()

    @r.get("")
    async def list_events(
        sinceMs: int | None = Query(None),
        untilMs: int | None = Query(None),
        deviceId: str | None = Query(None),
        eventType: str | None = Query(None),
        limit: int = Query(100),
    ):
        try:
            since = sinceMs if sinceMs is not None else int(time.time() * 1000) - 24 * 60 * 60 * 1000
            events = sqlite.query_events(
                since_ms=since, until_ms=untilMs,
                device_id=deviceId, event_type=eventType, limit=limit,
            )
            return {"ok": True, "events": [sqlite.event_to_dict(e) for e in events]}
        except Exception as e:
            return {"ok": False, "error": f"Failed to query events: {e}"}

    return r
